subprojects/core/src/main/java/org/gradle/plugin/use/internal/DefaultConfigurablePluginRequest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.plugin.use.internal;
+
+import org.gradle.api.Nullable;
+import org.gradle.plugin.use.PluginId;
+
+public class DefaultConfigurablePluginRequest implements InternalConfigurablePluginRequest {
+
+    private final InternalPluginRequest origonalRequest;
+    private boolean latch;
```

Comment:
Not necessary. If we create one of these, then it means the user is about to configure something in all cases.